tests/redirect.rs

```diff
@@ -59,197 +75,480 @@ pub async fn get_domain_redirect(user: &GooseUser) -> GooseTaskResult {
+                    "--expect-workers",
+                    &expect_workers.to_string(),
+                    "--users",
+                    &expect_workers.to_string(),
+                    "--hatch-rate",
+                    &expect_workers.to_string(),
+                ],
+            )
+        }
+    } else if worker.is_some() {
+        common::build_configuration(&server, vec!["--worker"])
+    } else {
+        if sticky {
+            common::build_configuration(&server, vec!["--sticky-follow"])
+        } else {
+            common::build_configuration(&server, vec![])
+        }
+    }
+}
+
+// Common validation for the load tests in this file.
+fn validate_redirect(test_type: &TestType, mock_endpoints: &Vec<MockRef>) {
+    match test_type {
+        TestType::Chain => {
+            // Confirm that we loaded the mock endpoints; while we never load the about page
+            // directly, we should follow the redirects and load it.
+            assert!(mock_endpoints[INDEX_KEY].times_called() > 0);
+            assert!(mock_endpoints[REDIRECT_KEY].times_called() > 0);
+            assert!(mock_endpoints[REDIRECT_KEY2].times_called() > 0);
+            assert!(mock_endpoints[REDIRECT_KEY3].times_called() > 0);
+            assert!(mock_endpoints[ABOUT_KEY].times_called() > 0);
+
+            // We should have called all redirects the same number of times as we called the
+            // final about page.
+            assert!(
+                mock_endpoints[REDIRECT_KEY].times_called()
+                    == mock_endpoints[REDIRECT_KEY2].times_called()
+            );
+            assert!(
+                mock_endpoints[REDIRECT_KEY].times_called()
+                    == mock_endpoints[REDIRECT_KEY3].times_called()
+            );
+            assert!(
+                mock_endpoints[REDIRECT_KEY].times_called()
+                    == mock_endpoints[ABOUT_KEY].times_called()
+            );
+        }
+        TestType::Domain => {
+            // Confirm that we load the index, about and redirect pages on the original domain.
+            assert!(mock_endpoints[SERVER1_INDEX_KEY].times_called() > 0);
+            assert!(mock_endpoints[SERVER1_REDIRECT_KEY].times_called() > 0);
+            assert!(mock_endpoints[SERVER1_ABOUT_KEY].times_called() > 0);
+
+            // Confirm that the redirect sends us to the second domain (mocked using a
+            // server on a different port).
+            assert!(mock_endpoints[SERVER2_INDEX_KEY].times_called() > 0);
+
+            // Confirm the we never loaded the about page on the second domain.
+            assert!(mock_endpoints[SERVER2_ABOUT_KEY].times_called() == 0);
+        }
+        TestType::Sticky => {
+            // Confirm we redirect on startup, and never load index or about.
+            assert!(mock_endpoints[SERVER1_INDEX_KEY].times_called() == 0);
+            // @FIXME: when https://github.com/tag1consulting/goose/issues/182 lands
```

Comment:
There is a [known Gaggle bug](https://github.com/tag1consulting/goose/issues/182) so I left a comment to fix this test when the bug is fixed.